src/client/producer.rs

```diff
@@ -435,7 +435,18 @@ where
     async fn flush(inner: &mut ProducerInner<A>, client: &dyn ProducerClient) {
         trace!("Flushing batch producer");
 
-        let (output, status_deagg) = inner.aggregator.flush();
+        let (output, status_deagg) = match inner.aggregator.flush() {
+            Ok(x) => x,
+            Err(e) => {
+                // Reset result slot
+                let slot = std::mem::take(&mut inner.result_slot);
+
+                slot.broadcast(AggregatedResult {
```

Comment:
It's personal preference and I know this PR didn't introduce it, but perhaps `AggregatedResult` could just be a type alias instead of a custom struct. This would then make this just `slot.broadcast(AggregatedError::Aggregator(Arc::new(e)))`

Reply:
done